### PR TITLE
feat(github): shared Shiki singleton + review-thread diff syntax highlighting (cave-4nfu)

### PR DESCRIPTION
**Pull request by BunsDev**
## Summary

Re-lands the shiki-highlighter extraction archived in the 2026-07-09 stash triage (bead **cave-4nfu**), rebased onto current main.

- **New `src/lib/shiki-highlighter.ts`** — one lazy app-wide Shiki singleton (theme + full `SHIKI_LANGS` grammar set), preserving main's cave-h1hi `structuredClone(moodCTheme)` fix that postdates the stash.
- **`message-bubble.tsx`** — drops its inline `createHighlighter` for the shared `getShikiHighlighter`; chat fences and the diff renderer can no longer double-load the WASM engine.
- **`gh-diff-view.tsx`** — `DiffHunk` gains token-level syntax highlighting: `path`'s extension picks the grammar, the hunk is highlighted as one document with meta rows blanked so tokens stay line-aligned, and the +/-/context marker sits in its own non-copyable `.gh-diff__marker` span so diff-strip coloring and token coloring compose. Unknown grammars skip the pass (plain render, no WASM load). Long lines wrap (`pre-wrap` + `overflow-wrap:anywhere`) instead of forcing horizontal scroll.
- **`github-view.tsx`** — review threads pass `thread.path` to `DiffHunk`.
- Pins: `gh-diff.test.ts` wiring section extended; `code-editor.test.ts` cave-h1hi clone pin follows the singleton into the lib (+ message-bubble must-not-own-an-instance pins).

**Deliberately not re-landed** from the stash: the chat-view `cancelSend` hunk — superseded by the S6 stream-health rework (cancelSend now issues an explicit `/api/chat/stop`; the stash's local-settle version would regress it).

## Verification
- `pnpm typecheck`, `pnpm lint`, `pnpm test:app` (884 files) green
- Prod build within bundle budgets (design-token drift gate satisfied via `tokenize-css.mjs`)
- Runtime probe: `codeToTokens` line alignment (blank lines → empty token arrays, meta rows blank)
- Real-app Playwright probe: `pre.shiki mood-c-dark` blocks with 5,250 inline-colored token spans through the shared singleton; zero page errors

Closes cave-4nfu.

### CODE REVIEW

File: src/components/gh-diff-view.tsx [hunk 1]
```diff
@@ -10,25 +13,64 @@ import { parseDiff, diffStats, diffLineClass } from "@/lib/gh-diff";
...
   hunk: string;
+  /** File path of the diff — its extension picks the Shiki grammar. */
+  path?: string | null;
   previewLines?: number;
   className?: string;
 }) {
   const lines = useMemo(() => parseDiff(hunk), [hunk]);
   const stats = useMemo(() => diffStats(lines), [lines]);
   const [expanded, setExpanded] = useState(false);
+  const lang = resolveShikiLang(path ? path.split(".").pop() : null);
```
[the remark this answers is not in the document]
Comment (BunsDev) [reply src/components/gh-diff-view.tsx]: Fixed in e45f46839 — `pathLangToken` extracts the basename first, then uses its extension when present or the bare name otherwise (`infra/Dockerfile` → `Dockerfile` → dockerfile grammar; `my.dir/Makefile` → `Makefile`). Behavior probed over the edge cases (subdir Dockerfile, dotted dirs, hidden files, trailing slash).

File: src/lib/gh-diff.test.ts [hunk 2]
```diff
@@ -51,10 +51,47 @@ assert.match(diffLineClass("context"), /gh-diff__line--ctx/, "context class");
...
+  diffView,
+  /import \{ getShikiHighlighter \} from "@\/lib\/shiki-highlighter"/,
+  "DiffHunk highlights through the shared Shiki singleton (no second createHighlighter instance)",
+);
+assert.doesNotMatch(diffView, /createHighlighter\(/, "DiffHunk must not create its own Shiki instance");
+assert.match(
+  diffView,
+  /resolveShikiLang\(path \? path\.split\("\."\)\.pop\(\) : null\)/,
+  "the diff grammar resolves from the file path's extension",
+);
```
[the remark this answers is not in the document]
Comment (BunsDev) [reply src/lib/gh-diff.test.ts:82]: Updated in e45f46839 — the pins now assert the basename-first split (`path.split("/").pop()`) and the bare-basename fallback in `pathLangToken`, matching the new resolution.
